test/common.py

```diff
@@ -1562,6 +1584,9 @@ def run_js(self, filename, engine=None, args=None,
       else:
         self.fail('JS subprocess failed (%s): %s (expected=%s).  Output:\n%s' % (error.cmd, error.returncode, assert_returncode, ret))
 
+    if run_in_tmpdir:
+      force_delete_contents(self.in_dir('fs'))
```

Comment:
In general tests do not need to (indeed should not since helps with debugging) cleanup after themselves like this.

Reply:
That's a good point! While there already is some existing cleanup logic it only ran after each "full" test, not in between each WASM engine run when iterating over them. This made the test fail because each test expects a blank directory.

What do you think about a subdir per each engine? So `out/temp/wasmer`, `out/temp/wasmtime`, etc.? Then the existing cleanup code should work just as is.

Reply:
> What do you think about a subdir per each engine? So out/temp/wasmer, out/temp/wasmtime, etc.? Then the existing cleanup code should work just as is.

I implemented this and it works very well!